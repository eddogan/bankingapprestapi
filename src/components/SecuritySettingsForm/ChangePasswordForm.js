import React, { Fragment, useState } from "react";
// Material-UI imports
import { Typography, Box, TextField, Button } from "@material-ui/core/";
import { makeStyles } from "@material-ui/core/styles";

export default function ChangePasswordForm() {
  const classes = useStyles();
  const [values, setValues] = useState("");

  const onSubmit = event => {
    event.preventDefault();
  };

  const handleInputChange = prop => event => {
    setValues({ ...values, [prop]: event.target.value });
  };

  return (
    <Fragment>
      <Box px={2} pb={2}>
        <Typography
          className={classes.moduleSubTitle}
          variant="h6"
          component="h1"
        >
          Change Password
        </Typography>
      </Box>
      <form onSubmit={onSubmit}>
        <Box px={2}>
          <TextField
            id="currentPassword"
            name="currentPassword"
            label="Current Password"
            variant="outlined"
            margin="normal"
            value=" "
            className="ajaxLabel"
            onChange={handleInputChange}
            fullWidth={true}
          />
        </Box>
        <Box px={2}>
          <TextField
            id="newPassword"
            name="newPassword"
            label="New password"
            variant="outlined"
            margin="normal"
            helperText="Your Password should include one uppercase letter, one special character, and should be at least 8 characters in length."
            value=" "
            className="ajaxLabel"
            onChange={handleInputChange}
            fullWidth={true}
          />
        </Box>
        <Box px={2}>
          <TextField
            id="verifyNewPassword"
            name="verifyNewPassword"
            label="Verify New Password"
            variant="outlined"
            margin="normal"
            value=" "
            className="ajaxLabel"
            onChange={handleInputChange}
            fullWidth={true}
          />
        </Box>
        <Box m={2} className={classes.aligncenter}>
          <Button variant="contained" color="primary" type="submit">
            Update password
          </Button>
        </Box>
      </form>
    </Fragment>
  );
}

const useStyles = makeStyles(theme => ({
  moduleSubTitle: {
    borderBottom: "1px dashed #eee"
  }
}));
