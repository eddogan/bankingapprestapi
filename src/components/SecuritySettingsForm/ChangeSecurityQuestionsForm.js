import React, { Fragment, useState, useEffect, useCallback } from "react";
import axios from "axios";
// Material-UI imports
import {
  Typography,
  Box,
  TextField,
  Button,
  OutlinedInput,
  FormControl,
  Select,
  InputLabel
} from "@material-ui/core/";
import { makeStyles } from "@material-ui/core/styles";
// Custom imports
import getApi from "../../helpers/GetApi";

export default function ChangeSecurityQuestionsForm() {
  const classes = useStyles();
  const [questions, setQuestions] = useState([]);
  const [clientQuestions, setClientQuestions] = useState({
    question1: "",
    question2: "",
    question3: "",
    answer1: "",
    answer2: "",
    answer3: ""
  });

  const inputLabel = React.useRef(null);
  const [labelWidth, setLabelWidth] = React.useState(0);
  React.useEffect(() => {
    setLabelWidth(inputLabel.current.offsetWidth);
  }, []);

  const fetchAllQuestions = useCallback(async () => {
    const securityQuestionsEndpoint = `${process.env.REACT_APP_API_URL}/useraccess/securityquestions`;
    const securityQuestionsResponse = await axios(
      getApi(securityQuestionsEndpoint)
    );
    const questions = Object.entries(securityQuestionsResponse.data).map(
      function(value, i) {
        return (
          <option key={i} value={value[0]}>
            {value[1]}
          </option>
        );
      }
    );
    setQuestions(questions);
  }, []);

  const fetchClientQuestions = useCallback(async () => {
    const securityQuestionsAnswersEndpoint = `${process.env.REACT_APP_API_URL}/useraccess/securityquestions_answers`;
    const securityQuestionsAnswersResponse = await axios(
      getApi(securityQuestionsAnswersEndpoint)
    );
    setClientQuestions({
      ...securityQuestionsAnswersResponse.data,
      question1: Object.entries(securityQuestionsAnswersResponse.data)[0][0],
      question2: Object.entries(securityQuestionsAnswersResponse.data)[1][0],
      question3: Object.entries(securityQuestionsAnswersResponse.data)[2][0],
      answer1: Object.entries(securityQuestionsAnswersResponse.data)[0][1],
      answer2: Object.entries(securityQuestionsAnswersResponse.data)[1][1],
      answer3: Object.entries(securityQuestionsAnswersResponse.data)[2][1]
    });
  }, []);

  useEffect(() => {
    fetchAllQuestions();
    fetchClientQuestions();
  }, [fetchAllQuestions, fetchClientQuestions]);

  const onSubmit = event => {
    event.preventDefault();
  };

  const handleChange = name => event => {
    setClientQuestions({
      ...clientQuestions,
      [name]: event.target.value
    });
  };

  return (
    <Fragment>
      <Box px={2} pb={2}>
        <Typography
          className={classes.moduleSubTitle}
          variant="h6"
          component="h1"
        >
          Manage security questions
        </Typography>
      </Box>
      <form onSubmit={onSubmit}>
        <FormControl className={classes.formControl} margin="normal">
          <InputLabel
            ref={inputLabel}
            htmlFor="question1NativeSimple"
            shrink={true}
            className="ajaxLabel"
          >
            Question #1
          </InputLabel>
          <Select
            native
            value={clientQuestions.question1}
            onChange={handleChange("question1")}
            input={
              <OutlinedInput
                name="question1"
                labelWidth={labelWidth}
                id="question1NativeSimple"
              />
            }
          >
            {questions}
          </Select>
        </FormControl>
        <Box px={2} pb={2}>
          <TextField
            id="answerOne"
            name="answerOne"
            label="Enter your answer for the first question"
            variant="outlined"
            margin="normal"
            helperText="Answers are case sensitive."
            fullWidth={true}
            value={clientQuestions.answer1}
            className="ajaxLabel"
            onChange={handleChange("answer1")}
          />
        </Box>
        <FormControl className={classes.formControl} margin="normal">
          <InputLabel
            ref={inputLabel}
            htmlFor="question2NativeSimple"
            shrink={true}
            className="ajaxLabel"
          >
            Question #2
          </InputLabel>
          <Select
            native
            value={clientQuestions.question2}
            onChange={handleChange("question2")}
            input={
              <OutlinedInput
                name="question2"
                labelWidth={labelWidth}
                id="question2NativeSimple"
              />
            }
          >
            {questions}
          </Select>
        </FormControl>
        <Box px={2} pb={2}>
          <TextField
            id="answerTwo"
            name="answerTwo"
            label="Enter your answer for the second question"
            variant="outlined"
            margin="normal"
            helperText="Answers are case sensitive."
            fullWidth={true}
            value={clientQuestions.answer2}
            className="ajaxLabel"
            onChange={handleChange("answer2")}
          />
        </Box>
        <FormControl className={classes.formControl} margin="normal">
          <InputLabel
            ref={inputLabel}
            htmlFor="question3NativeSimple"
            shrink={true}
            className="ajaxLabel"
          >
            Question #3
          </InputLabel>
          <Select
            native
            value={clientQuestions.question3}
            onChange={handleChange("question3")}
            input={
              <OutlinedInput
                name="question3"
                labelWidth={labelWidth}
                id="question3NativeSimple"
              />
            }
          >
            {questions}
          </Select>
        </FormControl>
        <Box px={2}>
          <TextField
            id="answerThree"
            name="answerThree"
            label="Enter your answer for the third question"
            variant="outlined"
            margin="normal"
            helperText="Answers are case sensitive."
            fullWidth={true}
            value={clientQuestions.answer3}
            className="ajaxLabel"
            onChange={handleChange("answer3")}
          />
        </Box>
        <Box m={2} className={classes.aligncenter}>
          <Button variant="contained" color="primary" type="submit">
            Update security questions
          </Button>
        </Box>
      </form>
    </Fragment>
  );
}

const useStyles = makeStyles(theme => ({
  moduleSubTitle: {
    borderBottom: "1px dashed #eee"
  },
  formControl: {
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
    "& label": {
      left: 24,
      top: "-8px",
      backgroundColor: "#fff",
      zIndex: 10,
      padding: "0 8px"
    }
  }
}));
