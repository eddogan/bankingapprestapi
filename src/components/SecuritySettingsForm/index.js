import React, { Fragment } from "react";
// Material-UI imports
import { Grid, Box } from "@material-ui/core";
// Custom imports
import ChangePasswordForm from "./ChangePasswordForm";
import ChangeSecurityQuestionsForm from "./ChangeSecurityQuestionsForm";

export default function SecuritySettingsForm() {
  return (
    <Fragment>
      <Box my={2}>
        <div className="moduleContent">
          <Grid
            container
            spacing={0}
            justify="space-between"
            alignItems="flex-start"
          >
            <Grid item xs={12} sm={12} md={12} lg={5}>
              <ChangePasswordForm />
            </Grid>
            <Grid item xs={12} sm={12} md={12} lg={6}>
              <ChangeSecurityQuestionsForm />
            </Grid>
          </Grid>
        </div>
      </Box>
    </Fragment>
  );
}
