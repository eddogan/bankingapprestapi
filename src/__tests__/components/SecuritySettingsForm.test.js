import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SecuritySettingsForm from "../../components/SecuritySettingsForm";

it("renders correctly", () => {
  const tree = renderer.create(<SecuritySettingsForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SecuritySettingsForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});
