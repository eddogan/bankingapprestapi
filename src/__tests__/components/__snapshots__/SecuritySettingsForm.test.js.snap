// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="moduleContent"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-5"
      >
        <div
          className="MuiBox-root MuiBox-root-107"
        >
          <h1
            className="MuiTypography-root makeStyles-moduleSubTitle-106 MuiTypography-h6"
          >
            Change Password
          </h1>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="MuiBox-root MuiBox-root-138"
          >
            <div
              className="MuiFormControl-root MuiTextField-root ajaxLabel MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="currentPassword"
              >
                Current Password
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-197"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="currentPassword"
                  name="currentPassword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=" "
                />
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-198"
          >
            <div
              className="MuiFormControl-root MuiTextField-root ajaxLabel MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="newPassword"
              >
                New password
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-197"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-describedby="newPassword-helper-text"
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="newPassword"
                  name="newPassword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=" "
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                id="newPassword-helper-text"
              >
                Your Password should include one uppercase letter, one special character, and should be at least 8 characters in length.
              </p>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-207"
          >
            <div
              className="MuiFormControl-root MuiTextField-root ajaxLabel MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="verifyNewPassword"
              >
                Verify New Password
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-197"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="verifyNewPassword"
                  name="verifyNewPassword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=" "
                />
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-208"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Update password
              </span>
            </button>
          </div>
        </form>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-6"
      >
        <div
          className="MuiBox-root MuiBox-root-231"
        >
          <h1
            className="MuiTypography-root makeStyles-moduleSubTitle-229 MuiTypography-h6"
          >
            Manage security questions
          </h1>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root makeStyles-formControl-230 MuiFormControl-marginNormal"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root ajaxLabel MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
              htmlFor="question1NativeSimple"
            >
              Question #1
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-197"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <select
                aria-invalid={false}
                className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
                disabled={false}
                id="question1NativeSimple"
                multiple={false}
                name="question1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-248"
          >
            <div
              className="MuiFormControl-root MuiTextField-root ajaxLabel MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="answerOne"
              >
                Enter your answer for the first question
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-197"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-describedby="answerOne-helper-text"
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="answerOne"
                  name="answerOne"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="answerOne-helper-text"
              >
                Answers are case sensitive.
              </p>
            </div>
          </div>
          <div
            className="MuiFormControl-root makeStyles-formControl-230 MuiFormControl-marginNormal"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root ajaxLabel MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
              htmlFor="question2NativeSimple"
            >
              Question #2
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-197"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <select
                aria-invalid={false}
                className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
                disabled={false}
                id="question2NativeSimple"
                multiple={false}
                name="question2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-249"
          >
            <div
              className="MuiFormControl-root MuiTextField-root ajaxLabel MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="answerTwo"
              >
                Enter your answer for the second question
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-197"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-describedby="answerTwo-helper-text"
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="answerTwo"
                  name="answerTwo"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="answerTwo-helper-text"
              >
                Answers are case sensitive.
              </p>
            </div>
          </div>
          <div
            className="MuiFormControl-root makeStyles-formControl-230 MuiFormControl-marginNormal"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root ajaxLabel MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
              htmlFor="question3NativeSimple"
            >
              Question #3
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-197"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <select
                aria-invalid={false}
                className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
                disabled={false}
                id="question3NativeSimple"
                multiple={false}
                name="question3"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-250"
          >
            <div
              className="MuiFormControl-root MuiTextField-root ajaxLabel MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="answerThree"
              >
                Enter your answer for the third question
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-196 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-197"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-describedby="answerThree-helper-text"
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="answerThree"
                  name="answerThree"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="answerThree-helper-text"
              >
                Answers are case sensitive.
              </p>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-251"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Update security questions
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
